indico/modules/oauth/blueprint.py

```diff
@@ -18,14 +18,31 @@
 
 from flask import request
 
-from indico.modules.oauth.controllers import RHOAuthUserProfile
+from indico.modules.oauth.controllers import (RHOAuthAuthorize, RHOAuthToken, RHOAuthUserProfile, RHOAuthAdmin,
+                                              RHOAuthAdminApplication, RHOAuthAdminApplicationNew,
+                                              RHOAuthAdminApplicationDelete, RHOAuthAdminApplicationReset,
+                                              RHOAuthAdminApplicationRevoke, RHOAuthUserTokenRevoke)
 from indico.web.flask.wrappers import IndicoBlueprint
 
+
 oauth_blueprint = _bp = IndicoBlueprint('oauth', __name__, template_folder='templates')
 
+# Application endpoints
+_bp.add_url_rule('/oauth/authorize', 'oauth_authorize', RHOAuthAuthorize, methods=('GET', 'POST'))
+_bp.add_url_rule('/oauth/token', 'oauth_token', RHOAuthToken, methods=('POST',))
+
+# Server administration
+_bp.add_url_rule('/admin/apps', 'apps', RHOAuthAdmin)
+_bp.add_url_rule('/admin/apps/new/', 'app_new', RHOAuthAdminApplicationNew, methods=('GET', 'POST'))
+_bp.add_url_rule('/admin/apps/<int:id>/', 'app_details', RHOAuthAdminApplication, methods=('GET', 'POST'))
+_bp.add_url_rule('/admin/apps/<int:id>/delete', 'app_delete', RHOAuthAdminApplicationDelete, methods=('POST',))
+_bp.add_url_rule('/admin/apps/<int:id>/reset', 'app_reset', RHOAuthAdminApplicationReset, methods=('POST',))
+_bp.add_url_rule('/admin/apps/<int:id>/revoke', 'app_revoke', RHOAuthAdminApplicationRevoke, methods=('POST',))
+
 # User profile
 with _bp.add_prefixed_rules('/user/<int:user_id>', '/user'):
-    _bp.add_url_rule('/oauth/', 'user_profile', RHOAuthUserProfile)
+    _bp.add_url_rule('/applications/', 'user_profile', RHOAuthUserProfile)
+    _bp.add_url_rule('/applications/<int:id>/revoke/', 'user_token_revoke', RHOAuthUserTokenRevoke, methods=('POST',))
 
 
 @_bp.url_defaults
```

Comment:
You mean that I have to add a check for the endpoint being `user_token_revoke` in the `_add_user_id` method?